src/shared/components/msktable/MSKTable.tsx

```diff
@@ -356,20 +387,42 @@ class MSKTableStore<T> {
                         {column.render(datum)}
                     </td>);
                 });
+                const rowProps:any = {};
+                if (this.dataStore.highlight(datum)) {
+                    rowProps.className = "highlight";
+                }
                 return (
-                    <tr>
+                    <tr {...rowProps}>
                         {tds}
                     </tr>
                 );
             });
     }
 
+    @action setFilterString(str:string) {
+        this.filterString = str;
+        this.page = 0;
+        this.dataStore.setFilter((d:T)=>{
+            let match = false;
+            for (const column of this.visibleColumns) {
+                match = (column.filter && column.filter(d, this.filterString, this.filterStringUpper, this.filterStringLower)) || false;
+                if (match) {
+                    break;
+                }
+            }
+            return match;
+        });
+    }
+
     @action setProps(props:MSKTableProps<T>) {
         this.columns = props.columns;
-        this.data = props.data;
         this._itemsLabel = props.itemsLabel;
         this._itemsLabelPlural = props.itemsLabelPlural;
         this._columnVisibility = this.resolveColumnVisibility(props.columns);
+
+        this.dataStore = props.dataStore || new MSKTableDataStore<T>(props.data || []);
+        this.dataStore.setSortAscending(this.sortAscending);
```

Comment:
this may cause data binding direction problems later if we have more than one component influencing sorting. thoughts?